src/routers/axelar/XCMRouter.sol

```diff
@@ -0,0 +1,205 @@
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) external;
+}
+
+interface AxelarGatewayLike {
+    function callContract(string calldata destinationChain, string calldata contractAddress, bytes calldata payload)
+        external;
+}
+
+contract ConnectorAxelarXCMRouter is AxelarExecutableLike {
+    address constant XCM_TRANSACTOR_V2_ADDRESS = 0x000000000000000000000000000000000000080D;
+
+    mapping(address => uint256) public wards;
+    //todo(nuno): do we really need this?
+    mapping(bytes32 => uint32) public executedCalls;
+
+    address public immutable centrifugeChainOrigin;
+    /// The origin of EVM -> Centrifuge messages; the trusted source origin of the Axelar-bridged
+    /// messages to be handled by this router.
+    address public axelarEVMRouterOrigin;
+    AxelarGatewayLike public immutable axelarGateway;
+    XcmWeightInfo internal xcmWeightInfo;
+
+    string public constant axelarCentrifugeChainId = "Centrifuge";
+    string public constant axelarCentrifugeChainAddress = "";
+
+    // --- Events ---
+    event Rely(address indexed user);
+    event Deny(address indexed user);
+    event File(bytes32 indexed what, XcmWeightInfo xcmWeightInfo);
+    event File(bytes32 indexed what, address addr);
+    event Executed(bytes32 indexed payload);
+
+    constructor(address centrifugeChainOrigin_, address axelarGateway_, address axelarEVMRouterOrigin_) {
+        centrifugeChainOrigin = centrifugeChainOrigin_;
+        axelarGateway = AxelarGatewayLike(axelarGateway_);
+        axelarEVMRouterOrigin = axelarEVMRouterOrigin_;
+        xcmWeightInfo = XcmWeightInfo({
+            buyExecutionWeightLimit: 19000000000,
+            transactWeightAtMost: 8000000000,
+            feeAmount: 1000000000000000000
+        });
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "ConnectorAxelarXCMRouter/not-authorized");
+        _;
+    }
+
+    modifier onlyCentrifugeChainOrigin() {
+        require(msg.sender == address(centrifugeChainOrigin), "ConnectorAxelarXCMRouter/invalid-origin");
+        _;
+    }
+
+    modifier onlyAxelarEVMRouterOrigin() {
+        require(
+            msg.sender == address(axelarEVMRouterOrigin),
+            "ConnectorAxelarXCMRouter/only-axelar-evm-router-origin-allowed"
+        );
+        _;
+    }
+
+    // --- Administration ---
+    function rely(address user) external auth {
+        wards[user] = 1;
+        emit Rely(user);
+    }
+
+    function deny(address user) external auth {
+        wards[user] = 0;
+        emit Deny(user);
+    }
+
+    function file(bytes32 what, address axelarEVMRouterOrigin_) external auth {
+        if (what == "axelarEVMRouterOrigin") {
+            axelarEVMRouterOrigin = axelarEVMRouterOrigin_;
+            emit File(what, axelarEVMRouterOrigin_);
+        } else {
+            revert("ConnectorXCMRouter/file-unrecognized-param");
+        }
+    }
```

Comment:
This brings up the question of who will be admin on this contract. Does this contract need a duplicate admin set up as EVM connectors now too?

Reply:
Good question, @AStox. To be discussed with @ilinzweilin and @offerijns 🔖

Reply:
we still have to figure out the admin setup on Moonbeam